src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -3,93 +3,136 @@
+    THROW_IF_FAILED(ShortAdd(selectionAnchorWithOffset.Y, _selectionVerticalOffset, &selectionAnchorWithOffset.Y));
+    THROW_IF_FAILED(ShortAdd(endSelectionPositionWithOffset.Y, _selectionVerticalOffset, &endSelectionPositionWithOffset.Y));
 
-    // clamp Y values to be within mutable viewport bounds
-    selectionAnchorWithOffset.Y = std::clamp(selectionAnchorWithOffset.Y, static_cast<SHORT>(0), _mutableViewport.BottomInclusive());
-    endSelectionPositionWithOffset.Y = std::clamp(endSelectionPositionWithOffset.Y, static_cast<SHORT>(0), _mutableViewport.BottomInclusive());
-
-    // clamp X values to be within buffer bounds
+    // clamp anchors to be within buffer bounds
     const auto bufferSize = _buffer->GetSize();
-    selectionAnchorWithOffset.X = std::clamp(_selectionAnchor.X, bufferSize.Left(), bufferSize.RightInclusive());
-    endSelectionPositionWithOffset.X = std::clamp(_endSelectionPosition.X, bufferSize.Left(), bufferSize.RightInclusive());
+    bufferSize.Clamp(selectionAnchorWithOffset);
+    bufferSize.Clamp(endSelectionPositionWithOffset);
 
     // NOTE: (0,0) is top-left so vertical comparison is inverted
-    const COORD& higherCoord = (selectionAnchorWithOffset.Y <= endSelectionPositionWithOffset.Y) ?
-                                   selectionAnchorWithOffset :
-                                   endSelectionPositionWithOffset;
-    const COORD& lowerCoord = (selectionAnchorWithOffset.Y > endSelectionPositionWithOffset.Y) ?
-                                  selectionAnchorWithOffset :
-                                  endSelectionPositionWithOffset;
-
-    selectionArea.reserve(lowerCoord.Y - higherCoord.Y + 1);
-    for (auto row = higherCoord.Y; row <= lowerCoord.Y; row++)
-    {
-        SMALL_RECT selectionRow;
+    // CompareInBounds returns whether A is to the left of (rv<0), equal to (rv==0), or to the right of (rv>0) B.
+    // Here, we want the "left"most coordinate to be the one "higher" on the screen. The other gets the dubious honor of
+    // being the "lower."
+    return bufferSize.CompareInBounds(selectionAnchorWithOffset, endSelectionPositionWithOffset) <= 0 ?
+               std::make_tuple(selectionAnchorWithOffset, endSelectionPositionWithOffset) :
+               std::make_tuple(endSelectionPositionWithOffset, selectionAnchorWithOffset);
+}
 
-        selectionRow.Top = row;
-        selectionRow.Bottom = row;
+// Method Description:
+// - constructs the selection row at the given row
+// NOTE: (0,0) is top-left so vertical comparison is inverted
+// Arguments:
+// - row: the buffer y-value under observation
+// - higherCoord: the physically "higher" coordinate (closer to the top-left)
+// - lowerCoord: the physically "lower" coordinate (closer to the bottom-right)
+// Return Value:
+// - the selection row needed for rendering
+SMALL_RECT Terminal::_GetSelectionRow(const SHORT row, const COORD higherCoord, const COORD lowerCoord) const
+{
+    SMALL_RECT selectionRow;

```

Comment:
nit: Even though each member of SMALL_RECT gets set currently it is best practice to always initialize a struct.   I am not sure what the team's stance is on this though.

 SMALL_RECT selectionRow{0};